modules/platforms/cpp/thin-client/src/impl/transactions/transactions_impl.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "impl/message.h"
+#include "impl/transactions/transactions_impl.h"
+#include "impl/response_status.h"
+
+using namespace ignite::common::concurrent;
+using namespace ignite::impl::thin;
+
+namespace ignite
+{
+    namespace impl
+    {
+        namespace thin
+        {
+            namespace transactions
+            {
+                TransactionImpl::TL_TXID TransactionImpl::threadTx;
+
+                TransactionsImpl::TransactionsImpl(const SP_DataRouter& router) :
+                    router(router)
+                {
+                }
+
+                template<typename ReqT, typename RspT>
+                void TransactionsImpl::SyncMessage(const ReqT& req, RspT& rsp)
```

Comment:
Can you re-name this method to something like `SendTxMessage`? SyncMessage should only be in network parts.